src/controllers/shows/tvgenres.js

```diff
@@ -1,139 +1,208 @@
-define(["layoutManager", "loading", "libraryBrowser", "cardBuilder", "lazyLoader", "apphost", "globalize", "appRouter", "dom", "emby-button"], function(layoutManager, loading, libraryBrowser, cardBuilder, lazyLoader, appHost, globalize, appRouter, dom) {
+define(["layoutManager", "loading", "libraryBrowser", "cardBuilder", "lazyLoader", "apphost", "globalize", "appRouter", "dom", "emby-button"], function (layoutManager, loading, libraryBrowser, cardBuilder, lazyLoader, appHost, globalize, appRouter, dom) {
     "use strict";
-    return function(view, params, tabContent) {
+
+    return function (view, params, tabContent) {
         function getPageData() {
-            var key = getSavedQueryKey(),
-                pageData = data[key];
-            return pageData || (pageData = data[key] = {
-                query: {
-                    SortBy: "SortName",
-                    SortOrder: "Ascending",
-                    IncludeItemTypes: "Series",
-                    Recursive: !0,
-                    EnableTotalRecordCount: !1
-                },
-                view: "Poster"
-            }, pageData.query.ParentId = params.topParentId, libraryBrowser.loadSavedQueryValues(key, pageData.query)), pageData
+            var key = getSavedQueryKey();
+            var pageData = data[key];
+
+            if (!pageData) {
+                pageData = data[key] = {
+                    query: {
+                        SortBy: "SortName",
+                        SortOrder: "Ascending",
+                        IncludeItemTypes: "Series",
+                        Recursive: true,
+                        EnableTotalRecordCount: false
+                    },
+                    view: "Poster"
+                };
+                pageData.query.ParentId = params.topParentId;
+                libraryBrowser.loadSavedQueryValues(key, pageData.query);
+            }
+
+            return pageData;
         }
 
         function getQuery() {
-            return getPageData().query
+            return getPageData().query;
         }
 
         function getSavedQueryKey() {
-            return libraryBrowser.getSavedQueryKey("seriesgenres")
+            return libraryBrowser.getSavedQueryKey("seriesgenres");
         }
 
         function getPromise() {
             loading.show();
             var query = getQuery();
-            return ApiClient.getGenres(ApiClient.getCurrentUserId(), query)
+            return ApiClient.getGenres(ApiClient.getCurrentUserId(), query);
         }
 
         function enableScrollX() {
-            return !layoutManager.desktop
+            return !layoutManager.desktop;
         }
 
         function getThumbShape() {
-            return enableScrollX() ? "overflowBackdrop" : "backdrop"
+            if (enableScrollX()) {
+                return "overflowBackdrop";
+            }
+
+            return "backdrop";
```

Comment:
this doesn't seem needed

Reply:
https://github.com/jellyfin/jellyfin-web/pull/509/commits/47cc029c7056b2ad334a93ba80f1568c6020a0ad